src/main/java/com/remondis/cdc/consumer/pactbuilder/ConsumerBuilderImpl.java

```diff
@@ -253,8 +266,16 @@ public PactDslJsonBody build(PactDslJsonBody pactDslJsonBody, T sampleData) {
   private BiFunction<PactDslJsonBody, Object, PactDslJsonBody> wrapInValueExtractor(PropertyDescriptor pd,
       BiFunction<PactDslJsonBody, Object, PactDslJsonBody> modifier) {
     BiFunction<PactDslJsonBody, Object, PactDslJsonBody> valueSupplierWrapper = (pactDslJsonBody, sampleValue) -> {
-      Object value = readOrFail(pd, sampleValue);
-      return modifier.apply(pactDslJsonBody, value);
+      try {
+        Object value = readOrFail(pd, sampleValue);
```

Comment:
The method `readOrFail` is currently doing this:
```
if (isNull(returnValue)) {
      throw new ConsumerBuilderException(
          "A property of the specified sample data object was null. The following get method returned null: "
              + readMethod.toGenericString());
    }
```
You can create something like a RuntimeException named `NoSampleValueException` and throw it instead of a `ConsumerBuilderException`. Then you could to something like
```suggestion
      } catch (NoSampleValueException e) {
        if (ignoreMissingValues) {
          return pactDslJsonBody;
        } else {
          throw e;
        }
      }
```
This would be better because if you do it with a `ConsumerBuilderException` you would suppress exceptions thrown by other modifiers due to other reasons.

Reply:
I don't really get it. The ConsumerBuilderException is (currently) only thrown in the readOrFail() method, so it will never be thrown if it is replaced by a NoSampleValueException, right?

I do see the added value in a more specific exception, but then I'd have it extend the ConsumerBuilderException class. I'll add an extra commit with my interpretation of your suggestion

Reply:
You're right: The `ConsumerBuilderException` will never be thrown currently. This exception type was initially designed for custom modifier like `PactDsl**Modifier` but I realized that it is actually unused.

Thanks for the extra commit. Looks nice!